### PR TITLE
feat: add option to remap URLs before opening

### LINKED ISSUE #8: Allow user to configure repo name
Currently opening a file in sourcegraph attempts to open `<url>/<host>/<filepath>` (ex: `http://localhost:1234/bitbucket.org/foo/bar/myfile`).

Repositories in sourcegraph are not necessarily configured with this naming scheme.

In fact in my local instance the repo name is just `foo/bar` so I must manually remove the `bitbucket.org/` segment each time.

### PR DESCRIPTION
**Pull request by mtbentley**
This is a naive solution to #8.  It adds a config option, which is a json dict where each key is replaced by the value in the sourcegraph url before opening it.  

For example, if you have git running at `git@git.example.org`, this fixes #8 by setting the `sourcegraph.remoteUrlReplacements` option to `{"git@git.example.org:": ""}`.

As another example, if `git@othergit.example.org:repo` should show up in sourcegraph as `src.example.org/othergit/repo`, you can add `"git@othergit.example.org:": "othergit/"`.

I am not super happy with this implementation, so I am open to suggestions for how to make the configuration easier to understand.

### CODE REVIEW

File: src/git.ts [hunk 1]
```diff
@@ -16,7 +17,14 @@ async function gitRemotes(repoDir: string): Promise<string[]> {
  */
 async function gitRemoteURL(repoDir: string, remoteName: string): Promise<string> {
     const { stdout } = await execa('git', ['remote', 'get-url', remoteName], { cwd: repoDir })
-    return stdout
+    const replacementsList = getRemoteUrlReplacements()
+    var remote = stdout
```
Comment (nicksnyder) [inline src/git.ts]: Can you remove this and just declare `stdout` to be a `var` instead of a `const`?

**Review by nicksnyder — COMMENTED** [review]
Thanks for suggesting an improvement!

@slimsag what do you think? Ideally this mapping would be done server side as part of the redirect, but if this PR improves the situation for now we could merge it and remove this config later if/when it becomes unnecessary.

**Comment by felixfbecker** [comment]
I noticed linting in CI didn't work correctly which is now fixed. @matthewbentley could you fix the ESLint violations?

**Review by felixfbecker — COMMENTED** [review]
The approach looks good to me

**Comment by fzxt** [comment]
This would be really nice to get in.

**Review by felixfbecker — APPROVED** [review]

**Comment by ShaneDelmore** [comment]
I wasn't able to locate the documentation on this feature. Can it be used to map any branch to master? I find when opening local branches that are not pushed to master, my link fails when 99% of the time what I really want is to fall back to master so I can send someone a link to existing code, not something I modified on my branch.

**Pull request merged**